CSI: Manager PublishVolume verify CSI node ID is not empty

olljanat opened the pull request:
**- What I did**
When I was working on to packaging DigitalOcean CSI driver for Swarm in https://github.com/olljanat/csi-plugins-for-docker-swarm/pull/15 I found that PublishVolume gets called before Node ID value is stored.

Returning error on this case looks to trigger Swarm logic to trying publish again until it works. 

However it looks to be that Docker engine need to be restarted once after plugin was installed before NodeID have value so there is more work need ton this area.

**- How I did it**
Implemented simple value check. Also added separate error message logging because it looks to be that error returned by `PublishVolume` does not appear to log.

**- How to test it**
Should be fine as long it passes CI.

**- Description for the changelog**

s4ke commented on manager/csi/plugin.go:
```diff
@@ -204,8 +207,13 @@ func (p *plugin) PublishVolume(ctx context.Context, v *api.Volume, nodeID string
 	}
+	csiNodeID := p.swarmToCSI[nodeID]
+	if csiNodeID == "" {
+		log.L.Errorf("CSI node ID is not yet registered. Plugin: %s , Swarm node ID: %s", p.name, nodeID)
```
should we maybe change the error message to something less specific here?

"CSI node ID not found for given Swarm node ID. Plugin: %s , Swarm node ID: %s"

or something? Then this error message will not be misleading if we run into it in the future.

s4ke commented on manager/csi/plugin.go:
```diff
@@ -204,6 +207,11 @@ func (p *plugin) PublishVolume(ctx context.Context, v *api.Volume, nodeID string
+	csiNodeID := p.swarmToCSI[nodeID]
+	if csiNodeID == "" {
+		log.L.Errorf("CSI node ID not found for given Swarm node ID. Plugin: %s , Swarm node ID: %s", p.name, nodeID)
+		return nil, status.Error(codes.FailedPrecondition, "CSI node ID is not yet registered")
```
i think the message in here should be the similar to the log message as well

s4ke commented:
Looks like CI is failing on exactly this precondition. I think the changes regarding error handling added here are correct, though.

olljanat commented:
Yeah. I simplified it a bit from original version so less tests was failing and fixed that one test set where one place node was called for `node1` and on another for `swarmNode1` so it failed on this precondition. Looks to be 🟢 now.

thaJeztah commented:
@neersighted @dperny ptal
